x-pack/solutions/search/plugins/enterprise_search/server/routes/enterprise_search/deprecations.ts

```diff
@@ -61,4 +62,29 @@ export function registerDeprecationRoutes({ router, log }: RouteDependencies) {
       });
     })
   );
+
+  router.post(
+    {
+      path: '/internal/enterprise_search/deprecations/set_enterprise_search_indices_read_only',
```

Comment:
Because this route now has another side effect: calling rollover on data streams (to create a new write index). There is a possibility of concurrency issues: what happens when this gets called 2-3 times quickly? We will end up with quite a few more rollovers than needed.

I think, overall, this risk is fairly low and to address this would be more complex. I just wanted to point out that even in a single Kibana situation this is currently not an atomic operation. Let me know what you think!

Reply:
Thanks for pointing this out - I think with a fix for the above (calling rollover on the same datastream name multiple times) will help alleviate this.